### ISSUE TITLE
F11 results in broken browser

### ISSUE THREAD

**Issue by dryeo** [bitwiseworks/mozilla-os2#218]
Pressing F11 to go into kiosk mode (full screen) results in a window with the browser window drawn the same size and the rest of the screen black. Pushing F11 again to restore the window results in a none responsive browser. The cursor does respond and change but scrolling, trying to change tabs etc results in no activity. Opening the hamburger menu results in a black area where the menus should be. On SeaMonkey the menus do work so the hamburger menu issue is a Firefox only problem.
The first time I experienced this, reopening Firefox and attempting to open a new tab resulted in the browser vanishing, no trp file and nothing in popup.log. After rebooting that issue went away.
Affects Firefox 45.5 and the latest SeaMonkey.

**Comment by LewisR** [comment]
Possibly related to #211

**Comment by dmik** [comment]
There are these new methods in nsWindow (at least on Windows) since ESR 45:
```
  virtual bool PrepareForFullscreenTransition(nsISupports** aData) override;
  virtual void PerformFullscreenTransition(FullscreenTransitionStage aStage,
                                           uint16_t aDuration,
                                           nsISupports* aData,
                                           nsIRunnable* aCallback) override;
```
A lack of those on OS/2 may be a reason for failures.

**Comment by dmik** [comment]
These new methods have something to do with some new full screen transition animation and they are disabled on Windows/Linux when no composition window manager is available. They are disabled on OS/2 as well now which totally makes sense as we don't have a composition manager. So the problem with fullscreen is somewhere else. This needs more investigation.

**Comment by dmik** [comment]
I found something interesting. Widgets on other platforms use so-called widget listeners to get notifications about various widget events (`mWidgetListener` and `mAttachedWidgetListener` fields of `nsBaseWidget`). However, OS/2 `nsWindow` code has never used them so far. Seems that it wasn't a problem in ESR 38 and before — perhaps ESR 45 uses listeners more intensively. Adding listener calls to resize event handlers fixes fullscreen work. Keyboard input remains broken though. I guess some more listener calls need to be added to fix it.

**Comment by dmik** [comment]
I was right. Notifying the widget listener in `nsWindow::MakeFullScreen` fixes both full screen and YouTube (#211) keyboard issues. And painting itself in fullscreen was fixed by adding `WillPaintWindow` and `DidPaintWindow` notifications in `nsWindow::OnPaint` which we also missed since long.
